### PR TITLE
Remove an allocation in read-only transactions.

### PR DESCRIPTION
**Pull request by petermattis**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3057)

### CODE REVIEW

**Comment by tschottdorf** [comment]
LGTM, but when was this code path actually exercised?

**Comment by petermattis** [comment]
We elide the end-transaction request if there were no writes in the transaction. For example, on read-only SQL operations.

File: client/db.go [hunk 1]
```diff
@@ -469,7 +469,7 @@ func (db *DB) Txn(retryable func(txn *Txn) error) error {
 // returns any errors.
```
Comment (bdarnell) [inline client/db.go]: Update the comment to say that it returns nil rather than an empty response for empty input.

Comment (petermattis) [inline client/db.go]: Done.

**Comment by bdarnell** [comment]
Special-casing a nil response seems like a very late and surprising place to handle this issue. Is there a good place to catch empty requests earlier? Otherwise LGTM.

**Comment by petermattis** [comment]
@bdarnell I could move this to `txn.send()`. Not sure that would be any clearer. Hard to do it earlier than that because the sequence of events is `txn.send(end-txn-request)`, `end-txn-request` gets elided because the txn has no writes, empty batch gets sent to `DB.send()`.

**Pull request merged**